packages/hooks/src/useEventListener.ts

```diff
@@ -17,17 +17,21 @@ interface UseEventOptions {
  * @param optional.dependencies Array to be passed to useEffect hook, such that the hook will only run if the array's values have changed.
  * @param optional.element Value to be passed as target of event handler, will default to document.
  */
-export default function useEventListener(
-  type: keyof GlobalEventHandlersEventMap,
-  eventCallback: (e) => void,
+export default function useEventListener<
+  Type extends keyof GlobalEventHandlersEventMap
+>(
+  type: Type,
```

Comment:
What's the benefit of making this generic?

Reply:
It allows TS to infer that if I pass in, say, `'click'`, then the event handler is going to be passed a `MouseEvent` rather than just a generic `Event`. So it'll have `clientX` and `clientY` and `button` and all that stuff on it. The built-in types for `addEventListener` are already doing the same thing, so I'm just kinda stealing from that.